### PR TITLE
Core/Auras: Implement SPELL_AURA_MOD_SPELL_COOLDOWN_BY_HASTE

### PR DESCRIPTION
**Pull request by efectokoleos**
**Changes proposed**:
- Implement aura SPELL_AURA_MOD_SPELL_COOLDOWN_BY_HASTE

**Target branch(es)**: 6x

**Tests performed**: Tested and working

### CODE REVIEW

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -12468,7 +12469,18 @@ void Unit::ApplyAttackTimePercentMod(WeaponAttackType att, float val, bool apply
         if (GetTypeId() == TYPEID_PLAYER)
         {
             if (att == BASE_ATTACK)
+            {
                 ApplyPercentModFloatValue(UNIT_FIELD_MOD_HASTE, val, !apply);
+
+                const Unit::AuraEffectList list = GetAuraEffectsByType(SPELL_AURA_MOD_SPELL_COOLDOWN_BY_HASTE);
+
+                for (Unit::AuraEffectList::const_iterator itr = list.begin(); itr != list.end(); ++itr)
```
Comment (Naios) [inline src/server/game/Entities/Unit/Unit.cpp]: You can simplify the two lines by using:
```c++
for (auto& effect : GetAuraEffectsByType(SPELL_AURA_MOD_SPELL_COOLDOWN_BY_HASTE))
```

Comment (Shauren) [inline src/server/game/Entities/Unit/Unit.cpp]: We don't use auto for non-iterator types, also AuraEffectList is a std::list<AuraEffect*>, there is no need to have a reference to pointer there

**Comment by Naios** [comment]
Looks good to me, anyone could test it?

**Comment by Aokromes** [comment]
Any reason to don't merge this?

**Comment by Naios** [comment]
No, I'll merge this later today.

**Comment by Shauren** [comment]
I just looked at how this aura is handled by the client and checked sniffs - your implementation is wrong.
No spellmod is used on retail for this (in fact you actually break it *clientside* by making it apply haste TWICE)

**Pull request closed without merging**